core/src/main/scala/kafka/log/Log.scala

```diff
@@ -197,6 +197,7 @@ class Log(val dir: File,
               warn("Found a corrupted index file, %s, deleting and rebuilding index...".format(indexFile.getAbsolutePath))
               indexFile.delete()
               segment.recover(config.maxMessageSize)
+              error("Corrupted Index File", e)
```

Comment:
If we already have a `warn` above, why do we need a separate `error`? If the exception and stack trace need to be logged, can we just incorporate it into the previous warning (and change to error if we think it needs to be at that log level)?